src/server/socksServer.ts

```diff
@@ -96,10 +96,16 @@ class SocksV5ServerParser {
     const socket = this._socket;
     let i = 0;
     const readByte = () => chunk[i++];
+    function assertAndClose(condition: unknown): asserts condition {
+      if (!condition) {
+        socket.end();
+        debugLogger.log('proxy', `Closed socket caused by false assertion: ${new Error().stack}`);
+      }
+    }
     while (i < chunk.length && this._phase !== ConnectionPhases.DONE) {
       switch (this._phase) {
         case ConnectionPhases.VERSION:
-          assert(readByte() === SOCKS_VERSION);
+          assertAndClose(readByte() === SOCKS_VERSION);
```

Comment:
I think you should also return from this function, or we'll get a lot of errors following the assertion, or perhaps even an infinite cycle.